### PR TITLE
(feat): org-roam-commit command

### PR DESCRIPTION
**Pull request by progfolio**
Add org-roam-commit to get current vc commit for org-roam.
Add org-roam-commit output to org-roam-diagnostics.

###### Motivation for this change
Should help with debugging issues

### CODE REVIEW

**Comment by progfolio** [comment]
I tried to be fairly guarded in the approach (inspired by magit-version), but I have not tested this on Windows. If the command fails it should just return nil and its output should not be included in `org-roam-diagnostics` output.

**Comment by nobiot** [comment]
@progfolio, happy to. Let me see if I can do this some time this week...

**Comment by nobiot** [comment]
Hi @progfolio, I tried `org-roam-commit` and `org-roam-diagnostics`.
See the results for the commit, nothing seems to have happened.
`(executable-find "git") finds the executable file,  so that's not the issue here.

E790FC29-DBF6-4EB6-9EB7-07660A0E9F19
**Figure 1.** Git animation for `org-roam-commit` 

image
**Figure 2.** Result from `org-roam-diagnostics` (chose Doom)

**Comment by progfolio** [comment]
Thank you for taking the time to test.
Are you familiar with using edebug to step through a function?

**Comment by nobiot** [comment]
Yes, and happy to do that. 
Let me try tomorrow -- it's late in my time zone (23:00).

File: org-roam.el [hunk 1]
```diff
@@ -1113,6 +1114,23 @@ Otherwise, behave as if called interactively."
...
+;;;###autoload
+(defun org-roam-commit (&optional message)
+  "Return the current commit of `org-roam' repository, nil if not found.
+Interactively, or when MESSAGE is non-nil, show in the echo area."
+  (interactive)
+  (when-let ((git (executable-find "git"))
+             (library (ignore-errors (file-chase-links (locate-library "org-roam.el")))))
+    (let* ((libdir (file-name-directory library))
+           (gitdir (expand-file-name ".git" (file-name-directory libdir))))
+      (when (and (file-exists-p gitdir)
```
Comment (nobiot) [inline org-roam.el:1126]: It's this condition. I use Doom, so use `straight`, and with cloned git repo that resides in my local machine.
This means.... The `.git` file does not exist in this location that the previous part finds: `C:\Users\nobiot\.emacs.d\.local\straight\build\org-roam\.git`

Comment (nobiot) [reply org-roam.el:1126]: "this condition" is meant to be `(and file-exists-p gitdir) -- sorry if the way I did this above was not clear. I guess commenting without using the review function would have been easier. Windows seems to correctly locate where the `.el` file for the package is. The issue seems to be related to how I use Emacs and packages.

**Comment by nobiot** [comment]
Sorry if the way I did this above was not clear. I guess commenting without using the review function would have been easier. Windows seems to correctly locate where the `.el` file for the package is.

File: org-roam.el (hunk 1 above)
Comment (progfolio) [reply org-roam.el:1126]: Thanks for looking into this.

I use straight as well, but on Linux.
On *nix systems, straight's build directory symlinks *.el files back to the original repo, so we find the correct directory. IIRC, that is not the case on Windows.

Comment (progfolio) [reply org-roam.el:1126]: https://github.com/raxod502/straight.el/blob/a7f94876b2bf96d2595706270be6630ecc94f0d3/straight.el#L692-L695

Looks like for Windows, straight copies the source files into the build dir.

**Comment by progfolio** [comment]
This will only work for a subset of users. Those using package.el will not have the git directory either.
Accounting for OS differences may be more trouble than it's worth, considering the limited audience.

@nobiot: Thanks for looking into this. I appreciate your help.

@jethrokuan: If having the commit is useful for us, the easiest way to implement something like this would involve us programmatically appending the hash to the Version string or storing it in our own metadata file that is guaranteed to ship w all package managers.

Going to put a freeze on this for now.

**Comment by jethrokuan** [comment]
> @jethrokuan: If having the commit is useful for us, the easiest way to implement something like this would involve us programmatically appending the hash to the Version string or storing it in our own metadata file that is guaranteed to ship w all package managers.

Yes, I think Magit does this in a Makefile, I've looked into it, but it seemed like a lot instrumentation. I think this is still useful for those who don't use package.el.

**Comment by progfolio** [comment]
Yeah, I was trying to think of  way to cleverly do it with a pre-commit hook.
I'll thnk on it some more.

**Pull request closed without merging**